feat(#6) 옷장 등록 페이지, 나의 옷장 페이지 구현

yhs-2551 opened the pull request:
## ISSUE
feat(#6) or UI(#6)

## Develop

**05.25 수정 부분**
- 피그마에서 사용한 값을 코드로 옮기면 한 화면에 안 담겨서 일부 변경(데스크탑, 100% 기준)
- 커멘트 언급 부분들 적용
- 프로그레스바 기준 위 아래 마진 70
- 반응형으로 하진 않았음
- 수정할 부분 있다면 리뷰 남겨 주시고, 저희 팀 빨리 백엔드 개발로 넘어가야 할 것 같네요

**옷장 등록 페이지**

- 이미지 업로드 방식: 파일 선택과 드래그 앤 드롭 방식 모두 지원
- 드랍다운 AI 분석 시: 분석 전에는 드롭다운에 “분석 전”으로 표시 및 선택 비활성화
- “대기 목록에 추가하기” 버튼: 표시/삭제 방식 대신 고정된 상태에서 활성화(Enable)/비활성화(Disable)로 처리(AI 분석 시 비활성화)
- 대기 목록 UI: 한 줄에 4개씩 2줄로 구성된 고정 높이 영역을 사용하면 한 화면에 모든 항목이 보이지 않아 외부 스크롤이 생기므로, UX상 접이식 형태로 구현
- 대기 목록 내부 스크롤: 피그마 디자인처럼 3줄 이상부터 내부 스크롤이 생김
- 아래 이미지들은 더미 데이터 사용

대기 목록에 아무것도 등록하지 않은 모습 
![image](https://github.com/user-attachments/assets/9c763878-0e62-4b7f-bb44-81d405a69f1f)

대기 목록에 이미지 10개 등록한 모습(등록된 데이터는 Ai 분석 전 더미 데이터, 대기 목록에 이미지 추가하면 외부 스크롤 발생)
![image](https://github.com/user-attachments/assets/5d48773d-575a-499c-af51-c5b646960dfe)


10개 등록후 대기 목록을 접은 모습
![image](https://github.com/user-attachments/assets/7cc56f91-cf13-4c0a-b4eb-e512694a282d)


**나의 옷장 페이지**

- 옷장 페이지에서 필터링 조회 시 디바운스 방식 고려 중

**더미 데이터 넣은 모습(전채선택, 초기화 버튼 추가)**
![image](https://github.com/user-attachments/assets/d6066a45-04bb-428b-89af-b33e23f4f5b3)


수정 모달 
 
![image](https://github.com/user-attachments/assets/a53482f7-2194-4ce7-a70e-784e3dadd0af)

- 초기 방문 시 개인 프로필, 의상 등록 상태 관리 논의 필요

**나의 옷장 페이지 -> 옷 추가하기 -> 옷 등록 페이지 **

![image](https://github.com/user-attachments/assets/72fa5044-72fa-46e6-8d23-153e2fe6cfb3)

**아래 App.jsx 부분 병합 후에 수정하면 됩니다**
```javascript

function App() {
    return (
        <>
            <BrowserRouter>
                <Routes>
                    {/* <Route path='/' element={<Home />} /> */}
                    {/* <Route path='/set-profile' element={<SetProfile />} /> */}
                    {/* <Route path='/my-profile' element={<MyProfile />} /> */}
                    <Route path='/my-closet' element={<MyClosetPage />} />
                    <Route path='/closet-add' element={<ClosetRegistrationPage showProgress={false} title='옷 등록' />} />
                    <Route path='/closet-registration' element={<ClosetRegistrationPage showProgress={true} title='FitU' />} />
                    <Route path='/' element={<div>Home Page</div>} />
                </Routes>
            </BrowserRouter>
        </>
    );
}

export default App;

``` 

## Test
Test 없음

yhs-2551 committed:
feat(#4): 옷장 등록, 옷장 조회 페이지 UI 구현

- 옷장 등록 페이지 UI 구현
  - 대기 목록 부분: 한 행에 4개씩 2줄로 구성된 고정 높이 영역을 사용하면 초기 화면에 외부 스크롤이 생겨서 UX상 접이식 형태로 구현
  - 이미지 업로드 방식: 파일 선택과 드래그 앤 드롭 방식 모두 지원
  - 대기 목록에 추가하기 버튼: 표시/삭제 방식 대신 고정된 상태에서 활성화(Enable)/비활성화(Disable)로 처리(AI 분석 시 비활성화)
  - 그 외 피그마와 동일
- 옷장 조회 페이지 UI 구현
    - 수정 모달 구현
    - 그 외 피그마와 동일
- 전역 상수 처리를 위해 constants 폴더 추가

yhs-2551 committed:
feat(#6): 수정 모달 이중 스크롤 현상 수정, 전체 선택 및 초기화 버튼 추가

rosieisor reviewed (approved):
확인했습니다. 고생 많으셨습니다.

ottozip commented on src/pages/ClosetRegistrationPage/ClosetRegistrationPage.jsx:
```diff
@@ -0,0 +1,199 @@
+            <main className='max-w-4xl mx-auto flex flex-col w-full flex-1'>
+                {/* 헤더 밑부분 영역 */}
+                <div className='mb-3'>
+                    <h1 className='text-2xl font-bold text-center text-black mb-1'>FitU</h1>
```
스타일은 피그마와 차이가 있는 부분만 말씀 드리겠습니다.

1. 타이틀 폰트 크기 32px (2rem)
2. 상단 마진 70px (rem으로 나중에 변환)

전체적으로 통일만 되면 어느 스타일로 하든 상관 없어서 회의 때 의견 나누면 좋을 것 같습니다!

yhs-2551 replied:
피그마에 나와 있는 값들로 디자인하면 실제 구현 했을 때 피그마 처럼 레이아웃이 안나옵니다. 또한 UI상 여백이 너무 크며 버튼의 크기도 큰 느낌이 있습니다.
일단은 통일성은 위해서 맞춰 놓았고, 최근 회의 때 스타일에 대한 추가 얘기는 없었는데 추후 스타일 관련해서 수정 사항이 있다면 회의때 말씀 부탁드려요.

ottozip commented on src/pages/ClosetRegistrationPage/ClosetRegistrationPage.jsx:
```diff
@@ -0,0 +1,199 @@
+                </div>
+                {/* 옷 등록 영역 - 단일 컨테이너로 구성 */}
+                <div className='bg-white shadow-lg rounded-xl pt-8'>
+                    {/* 내용 영역: 이미지 업로더와 속성 선택기를 포함한 그리드 */}
+                    <div className='grid grid-cols-1 md:grid-cols-2 items-stretch'>
+                        {/* 왼쪽: 이미지 업로더 */}
+                        <div className='flex justify-center md:justify-end mb-4 md:mb-0 pr-2 md:pr-8'>
+                            <div className='max-w-xs'>
+                                <ImageUploader
+                                    ref={imageUploaderRef}
+                                    uploadedImage={uploadedImage}
+                                    onImageUpload={handleImageUpload}
+                                    onImageRemove={handleImageRemove}
+                                />
+                            </div>
+                        </div>
+
+                        {/* 오른쪽: 속성 선택기 */}
+                        <div className='flex flex-col pl-2 md:pl-8'>
+                            <div>
+                                <AttributeSelectors attributes={attributes} onAttributeChange={handleAttributeChange} isAnalyzed={isAnalyzed} />
+                            </div>
+                        </div>
+                    </div>
+
+                    {/* "대기 목록에 추가하기" 버튼 - 가운데 배치 */}
+                    <div className='flex justify-center'>
+                        <button
+                            type='button'
+                            onClick={handleAddToWaitlist}
+                            // disabled={!canAddToWaitlist}
+                            className='w-[200px] h-[36px] bg-black hover:bg-gray-800 mb-8 mt-8 text-white font-semibold py-2 px-4 rounded-md text-xs
+                            disabled:bg-gray-300 disabled:text-gray-500 disabled:cursor-not-allowed cursor-pointer
+                            transition-colors duration-150 ease-in-out'
+                        >
+                            {/* {isAnalysisInProgress ? "분석 중..." : "대기 목록에 추가하기"} */}
+                            대기 목록에 추가하기
+                        </button>
+                    </div>
+                </div>
```
1. 이미지 업로더와 속성 선택기 사이가 너무 가까움
2. 대기 목록에 추가하기 버튼은 높이 45px
3. 대기 목록에 추가하기 버튼의 텍스트 사이즈 16px

hover 스타일은 다음 회의 때 의견 나눠 보고 싶습니다!

ottozip commented on src/pages/ClosetRegistrationPage/ClosetRegistrationPage.jsx:
```diff
@@ -0,0 +1,199 @@
+                    </div>
+                </div>
+                {/* 하단 네비게이션 버튼 - 가운데 정렬 */}
+                <div className='flex justify-center space-x-4 py-2'>
+                    <button
+                        type='button'
+                        className='px-8 py-2 border border-black rounded text-xs font-medium text-black cursor-pointer hover:bg-gray-100 focus:outline-none'
+                    >
+                        이전
+                    </button>
+                    <button
+                        type='button'
+                        className='px-8 py-2 bg-black hover:bg-gray-800 text-white cursor-pointer rounded text-xs font-medium focus:outline-none'
+                    >
+                        다음
+                    </button>
+                </div>
```
1. 버튼 높이 45px
2. 버튼 텍스트 사이즈 16px
3. 이전 버튼 테두리 색상 #828282, 100%

ottozip commented on src/pages/ClosetRegistrationPage/Waitlist.jsx:
```diff
@@ -0,0 +1,27 @@
+        <div className='w-full h-full'>
+            {items.length === 0 ? (
+                <div className='h-full w-full flex items-center justify-center'>
+                    <p className='text-center text-sm text-gray-500'>아직 옷장에 넣을 옷이 없어요.</p>
```
1. 텍스트 사이즈 16px
2. 텍스트 색상 #828282, 100%

ottozip commented on src/pages/MyClosetPage/MyClosePage.jsx:
```diff
@@ -0,0 +1,117 @@
+        <div className='min-h-screen bg-[#F7F7F7] py-6 px-4 sm:px-6 lg:px-8'>
+            <main className='max-w-6xl mx-auto flex flex-col'>
+                <div className='mb-8 w-full'>
+                    <h1 className='text-3xl font-bold text-center text-black mb-6'>나의 옷장</h1>
```
1. 텍스트 사이즈 32px
2. 상단 마진 70px

ottozip commented on src/pages/MyClosetPage/MyClosePage.jsx:
```diff
@@ -0,0 +1,117 @@
+                            <ClosetFilter onFilterChange={handleFilterChange} />
+                        </div>
+                        {/* 옷 추가하기 버튼은 오른쪽에 */}
+                        <button
+                            onClick={handleAddItem}
+                            className='flex items-center bg-black text-white text-xs font-semibold px-3 py-2 rounded-md hover:bg-gray-800 transition-colors whitespace-nowrap cursor-pointer'
+                        >
+                            <PlusIcon className='w-4 h-4 mr-1 fill-white stroke-white' />옷 추가하기
+                        </button>
```
1. 버튼 높이 45px
2. 버튼의 텍스트 사이즈 16px

ottozip commented on src/pages/MyClosetPage/MyClosePage.jsx:
```diff
@@ -0,0 +1,117 @@
+                        </div>
+                    </div>
+                ) : (
+                    <div className='bg-white p-12 text-center text-gray-500 shadow-lg rounded-xl'>
+                        <p className='text-gray-500 text-lg mb-2'>옷장이 비어있습니다</p>
+                        <p className='text-gray-400 text-sm mb-4'>옷을 추가하여 나만의 옷장을 채워보세요</p>
+                    </div>
```
이 부분은 피그마에 없는 부분이라 의견을 나눠봐야겠지만 비슷한 부분에 대해 피그마에서는 다음 규칙으로 디자인 했습니다.

1. 텍스트 사이즈 16px
2. 텍스트 색상 #828282, 100%

다음 회의 때 의견 나눠보면 좋을 것 같습니다.

ottozip reviewed (commented):
확인했습니다. 수고 많으셨습니다!
일단 스타일 관련된 사항 위주로 피드백 남겼습니다. 지륜님 PR에도 의견 나눌 것들 남겨뒀으니 참고해서 다음 회의 때 이야기 나눠보면 좋을 것 같습니다.

rosieisor commented:
`헤더`와 `FitU 문구`, `프로그레스 바` 아래와 같이 적용해주시면 됩니다. 

```javascript
import Header from "../components/Header";
import ProgressBar from "../components/ProgressBar/ProgressBar";

const page = () => {
  return (
    <div className="min-h-screen bg-[#F7F7F7]">
      <Header />
      <h1 className="pt-[7.5rem] text-[2rem] font-bold text-center">FitU</h1>
      <ProgressBar activeStep={2} />
      {/* ... */}
    </div>
  );
}
```

yhs-2551 committed:
feat(#6): 초기 옷장 등록 페이지, 나의 옷장 페이지, 옷 등록 페이지 수정

- 첫 방문 시 옷장 등록 페이지 프로그레스바 추가
- 나의 옷장 페이지에서 옷 추가하기 클릭 시 옷 등록 페이지 추가(ProgressBar 제외)
- 각 페이지 스타일 수정

yhs-2551 committed:
feat(#6): refactor: 대기 목록에 아이템 추가 시 이전, 다음 버튼 마진값 조정

yhs-2551 committed:
feat(#6): [refeactor]: 프로그레스바 상단 부분 마진 값 조정

yhs-2551 committed:
feat/6 [refactor]: 대기 목록 텍스트 부분 마진 값 조정

yhs-2551 commented:
- 피그마에서 사용한 값을 코드로 옮기면 한 화면에 안 담겨서 일부 변경(데스크탑, 100% 기준)
- 커멘트 언급 부분들 적용 
- 프로그레스바 기준 위 아래 마진 70
- 반응형으로 하진 않았음 
- 수정할 부분 있다면 리뷰 남겨 주시고, 저희 팀 빨리 백엔드 개발로 넘어가야 할 것 같네요

yhs-2551 committed:
feat(#6): [feat] 옷 등록 페이지 유효성 검사 추가

rosieisor committed:
Merge branch 'develop' into feat/6

yhs-2551 committed:
feat(#6): 전역 상태 관리 추가, 사용자 프로필 페이지 일부 수정

- 프로그레스바 위 아래 마진 값 사용자 페이지, 옷 등록 페이지 통일(위: 50px, 아래: 70px)

yhs-2551 committed:
feat(#6): 프론트, 백엔드 연동 구현 완료

- 초기 방문 시 사용자 정보, 의상 정보 등록 후 최종 등록 완료 -> 로컬스토리지에 userId 반환
- 유효성 검사: 의상 정보 등록 시 이미지 업로드 필수, 모든 속성 선택 필수, 상의 및 하의(한 쌍) / 원피스 대기 목록에 추가 필수
- 사용자 의상 정보 페이지 조회, 필터 기반 조회, 수정 모달(유효성 검사는 의상 정보 등록 때와 동일), 삭제 연동 완료

- 추후 AI 분석 결과값 매핑 필요
